infor-design/enterprise
# 841 - calendar features and related issues

**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Working on features to add to the calendar #841.
During the process moved keyboard code and verified it for related failing tests and umalqura calendar.
Did some cleanup work while testing the components effected (4 components on #1136)
Fixed the handling of Arabic to be more accurate and not fail on months with different numbers of days.
Moved keyboard logic

**Related github/jira issue (required)**:
Closes #841 
Closes #1189 
Closes #1197 

**Steps necessary to review your pull request (required)**:

841 -Calendar Features 
http://localhost:4000/components/calendar/example-index.html

- peep the events view on the right, clicking the events will select them in the page, it shows 14 days in advance (configurable)
- all text in the events are now left aligned
- selecting a day from the month switcher datepicker now goes to that direct day.
- RTL has been cleaned up http://localhost:4000/components/calendar/example-index.html?locale=ar-SA
- can now tab into the calendar and use arrow keys to keyboard around on the calendar days

1189 - Tests
These failing tests are now back in (were xitted) , they should pass

1197 - Range Bug
- Go to http://localhost:4000/components/datepicker/example-range.html
- Click on the picker and select a range
- Go back in the field
- Clear the values
- Open again

(note this was fixed because the keyboard stuff meant refactoring so had to verify this functionality)
(note 2 only 1 part is fixed the other part is a duplicate)

Impact
Do some testing on datepicker. Particularly Arabic and keyboard
Added many new tests.
